test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -2951,4 +2951,31 @@ TEST_P(ExtProcIntegrationTest, SkipHeaderTrailerSendBodyClientSendAll) {
   verifyDownstreamResponse(*response, 200);
 }
 
+TEST_P(ExtProcIntegrationTest, SendBodyBufferedPartialWithTrailer) {
```

Comment:
could you in the test add a buffer limit and verify how much data is sent to the ext_proc server in parital mode?

Reply:
Please check above comment. Overall, if the test send the data over the limit, then the added code in this PR will not be hit since there will be no data left in the chunk_queue_.

The original problem is due to in the BUFFERED_PARTIAL mode,  there is data in the chunk_queue_, but never hit the limit, then the trailer arrives.